src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1090,6 +1091,53 @@ class spell_warr_sword_and_board : public SpellScriptLoader
         }
 };
 
+// 215538 - Trauma
+// Updated 7.1.5
+class spell_warr_trauma : public SpellScriptLoader
+{
+public:
+    spell_warr_trauma() : SpellScriptLoader("spell_warr_trauma") { }
+
+    class spell_warr_trauma_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warr_trauma_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_TRAUMA_EFFECT
+            });
+        }
+
+        void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+            if (Unit* caster = GetCaster())
+            {
+                if (Unit* target = eventInfo.GetActionTarget())
+                {
+                    //Get the Remaining Damage from the aura (if exist)
+                    int32 remainingDamage = target->GetRemainingPeriodicAmount(target->GetGUID(), SPELL_WARRIOR_TRAUMA_EFFECT, SPELL_AURA_PERIODIC_DAMAGE);
+                    //Get 25% of damage from the spell casted (Slam & Whirlwind) plus Remaining Damage from Aura
+                    int32 damage = (eventInfo.GetDamageInfo()->GetDamage() * 25.0 / 100.0) + remainingDamage;
```

Comment:
Don't hardcode the 25.0.. 

replace 25.0 with aurEff->GetAmount()

and for the 100.0, replace it with 100.0f